pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go

```diff
@@ -0,0 +1,323 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	testutil "k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state/testing"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+)
+
+const testingDir = "/tmp"
+
+func TestCheckpointStateRestore(t *testing.T) {
+	testCases := []struct {
+		description       string
+		checkpointContent string
+		policyName        string
+		expectedError     string
+		expectedState     *stateMemory
+	}{
+		{
+			"Restore non-existing checkpoint",
+			"",
+			"none",
+			"",
+			&stateMemory{},
+		},
+		{
+			"Restore default cpu set",
+			`{
+				"PolicyName": "none",
```

Comment:
Please add JSON encoding annotations so that we can read checkpoint files written by the old file-backed state implementation.

e.g.

```json
{
  "policyName": "none",                                                   
  "defaultCpuSet": "23-24",                                               
  "entries": {
    "container1": "4-6",
    "container2": "1-3"
  }
}
```

Looks like the structure is good, we just have a problem with letter-case. A tougher problem may be the checksum field, which is new to the checkpoint format. Maybe we can relax the parser such that it can tolerate reading a checkpoint with missing checksum but always write the checksum back. Ideally we should be able to support a transparent upgrade from file-backed state checkpoints to checkpoint manager without requiring a node drain.

To be sure, let's add an internal roundtrip test to verify that checkpoints written by state_file can be read by state_checkpoint.